daffodil-tdml-lib/src/test/scala/org/apache/daffodil/tdml/scalaxb/ScalaxbTests.scala

```diff
@@ -0,0 +1,142 @@
+        </tdml:document>
+      )
+    val infoset =
+      scalaxb.fromXML[InfosetType](
+        <tdml:infoset xmlns:tdml="http://www.ibm.com/xmlns/dfdl/testData">
+          <tdml:dfdlInfoset>
+            <ipv4:IPv4Header>
+              <Version>4</Version>
+              <IHL>5</IHL>
+              <DSCP>0</DSCP>
+              <ECN>0</ECN>
+              <Length>115</Length>
+              <Identification>0</Identification>
+              <Flags>2</Flags>
+              <FragmentOffset>0</FragmentOffset>
+              <TTL>64</TTL>
+              <Protocol>17</Protocol>
+              <Checksum>47201</Checksum>
+              <IPSrc>C0A80001</IPSrc>
+              <IPDest>C0A800C7</IPDest>
+              <FakeData>115</FakeData>
+              <ComputedChecksum>47201</ComputedChecksum>
+            </ipv4:IPv4Header>
+          </tdml:dfdlInfoset>
+        </tdml:infoset>
+      )
+    val expectedParserTestCase = 
+      scalaxb.fromXML[ParserTestCaseType](
+        <tdml:parserTestCase name="IPv4_1" root="IPv4Header" model="org/apache/daffodil/layers/IPv4.dfdl.xsd"
+    roundTrip="none" xmlns:tdml="http://www.ibm.com/xmlns/dfdl/testData" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/"
+  xmlns:xs="http://www.w3.org/2001/XMLSchema"
+  xmlns:fn="http://www.w3.org/2005/xpath-functions"
+  xmlns:dfdlx="http://www.ogf.org/dfdl/dfdl-1.0/extensions"
+  xmlns:ipv4="urn:org.apache.daffodil.layers.IPv4"
+  xmlns:chksum="urn:org.apache.daffodil.layers.IPv4Checksum"
+  xmlns:ex="http://example.com"
+  xmlns:tns="http://example.com">
+          <tdml:document>
+            <tdml:documentPart type="byte">{scala.xml.PCData("4500 0073 0000 4000 4011 b861 c0a8 0001 c0a8 00c7")}</tdml:documentPart>
+          </tdml:document>
+          <tdml:infoset>
+            <tdml:dfdlInfoset>
+              <ipv4:IPv4Header>
+                <Version>4</Version>
+                <IHL>5</IHL>
+                <DSCP>0</DSCP>
+                <ECN>0</ECN>
+                <Length>115</Length>
+                <Identification>0</Identification>
+                <Flags>2</Flags>
+                <FragmentOffset>0</FragmentOffset>
+                <TTL>64</TTL>
+                <Protocol>17</Protocol>
+                <Checksum>47201</Checksum>
+                <IPSrc>C0A80001</IPSrc>
+                <IPDest>C0A800C7</IPDest>
+                <FakeData>115</FakeData>
+                <ComputedChecksum>47201</ComputedChecksum>
+              </ipv4:IPv4Header>
+            </tdml:dfdlInfoset>
+          </tdml:infoset>
+        </tdml:parserTestCase>
+      )
+    val actualParserTestCase: ParserTestCaseType =
+      testSuite
+        .testSuiteChoices
+        .headOption
+        .map(_.as[ParserTestCaseType])
+        .getOrElse(throw new AssertionError("first test case is missing")) // fail() returns Unit so mimic JUnit
+    assertEquals(expectedParserTestCase, actualParserTestCase)
```

Comment:
This test case also fails, but it is going to take some time to figure out what is different. I suspect it some kind of namespace problem....